src/scripts/components/Auth.js

```diff
@@ -1,19 +1,95 @@
 import User from '@models/User';
+import AuthService from '@services/AuthService';
+import emitter from '@services/EventEmitter';
 
 class Auth {
   constructor() {
     this.authorization = document.getElementById('authorization');
+    this.dashboard = document.getElementById('dashboard');
+
     this.signinForm = document.getElementById('sign-in');
+    this.signinUsername = document.getElementById('sign-in__username');
+    this.signinPassword = document.getElementById('sign-in__password');
+
     this.signupForm = document.getElementById('sign-up');
+    this.signupUsername = document.getElementById('sign-up__username');
+    this.signupEmail = document.getElementById('sign-up__email');
+    this.signupPassword = document.getElementById('sign-up__password');
+
+    this.toSignIn = document.getElementById('to-sign-in');
+    this.toSignUp = document.getElementById('to-sign-up');
 
     this.render = this.render.bind(this);
+    this.registerListener = this.registerListener.bind(this);
+
+    this.showAuth = this.showAuth.bind(this);
+    this.hideAuth = this.hideAuth.bind(this);
+
+    this.signInFormSubmit = this.signInFormSubmit.bind(this);
+    this.signUpFormSubmit = this.signUpFormSubmit.bind(this);
+
+    this.changeToSignIn = this.changeToSignIn.bind(this);
+    this.changeToSignUp = this.changeToSignUp.bind(this);
+
+    this.registerListener();
   }
 
   render() {
-    if (User.token) {
-      this.authorization.classList.add('hide');
-    } else {
-      this.authorization.classList.remove('hide');
-    }
+    User.token
+      ? this.hideAuth()
+      : this.showAuth();
+  }
+
+  showAuth() {
+    this.authorization.classList.remove('hide');
+    this.dashboard.classList.add('hide');
+  }
+
+  hideAuth() {
+    this.authorization.classList.add('hide');
+    this.dashboard.classList.remove('hide');
+  }
+
+  signInFormSubmit(e) {
+    e.preventDefault();
+
+    AuthService.Login({
+      identifier: this.signinUsername.value,
+      password: this.signinPassword.value,
+    });
+
+    emitter.subscribe('authorized', this.render);
```

Comment:
You can subscribe to the event only once in the registerListener() method.

Reply:
What about subscribing to event listener in App once?

Reply:
Fine, then we can add the Dashboard render to the callback.

Reply:
Done. Moved to App file.